### PR TITLE
Update PVGIS docs and test to comply with PVGIS version 5.2

### LINKED ISSUE #1415: Changes to PVGIS in version 5.2
**Describe the bug**
PVGIS is releasing version 5.2 on March 1st, 2022. In the new version, the irradiance databases CMSAF and COSMO have been removed. Additionally, the ERA5 coverage has been extended to cover almost the entire globe and a new database, SARAH-2, has been added.

The full description can be found [here](https://joint-research-centre.ec.europa.eu/pvgis-photovoltaic-geographical-information-system/pvgis-releases/pvgis-52_en).

The new default databases can be seen here:
image

**Additional context**
There is nothing that indicates any changes to the API, so the pvlib.iootools functions ``get_pvgis_tmy`` and ``get_pvgis_hourly`` should continue to work.

However, some of the test files used in `test_pvgis.py` are from the CMSAF database. Since those tests seem to be either based on local files or mock tests, I do not expect test failures. However, it would probably make sense to update the tests regardless.

### PR DESCRIPTION
**Pull request by AdamRJensen**
- [x] Closes #1415
 - [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [x] Tests added
 - ~~[ ] Updates entries in [`docs/sphinx/source/reference`](https://github.com/pvlib/pvlib-python/blob/master/docs/sphinx/source/reference) for API changes.~~
 - ~~[ ] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/master/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).~~
 - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels (including `remote-data`) and Milestone are assigned to the Pull Request and linked Issue.

This PR removes references to the retired PVGIS irradiance databases CMSAF and COSMO. The CMSAF test file has been replaced with a SARAH2 test file. Additionally, minor doc updates were made, e.g., `PVGIS_VARIABLE_MAP` was renamed to `VARIABLE_MAP`.

### CODE REVIEW

**Comment by AdamRJensen** [comment]
There's hardly anything to add to the whatsnew as the functionality has not changed. If anyone thinks that it would be meaningful I'll be happy to add an entry though.

File: pvlib/tests/iotools/test_pvgis.py
Comment (kanderso-nrel) [inline pvlib/tests/iotools/test_pvgis.py:36]: How come the in-plane components got dropped here?  It seems like the API still returns them -- compare this from the test file:
```json
{"time": "20130101:0010", "P": 0.0, "G(i)": 0.0, "H_sun": 0.0, "T2m": -0.97, "WS10m": 1.52, "Int": 0.0}
```
versus this from the API itself (using the `v5_2` url):
```json
{"time": "20130101:0010", "P": 0.0, "Gb(i)": 0.0, "Gd(i)": 0.0, "Gr(i)": 0.0, "H_sun": 0.0, "T2m": -0.97, "WS10m": 1.52, "Int": 0.0}
```

Comment (AdamRJensen) [reply pvlib/tests/iotools/test_pvgis.py:36]: I had generated the file with `components=0` but not updated the test instances to match. I guess this is the flipside of mock testing... 

Since the affected tests were only mock test, there is no real connection with what is actually being requested and the passed test file.

**Review by kanderso-nrel — APPROVED** [review]
LGTM, thanks @AdamRJensen!

**Comment by AdamRJensen** [comment]
@pvlib/pvlib-maintainer feel free to leave a second review, otherwise, I'll go ahead and merge this PR on August 12th.

**Pull request merged**
